services/src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -216,22 +228,39 @@ public Thread newThread(Runnable r) {
   private final String _topic;
   private final double _partitionsToBrokerRatio;
   private final int _scheduleIntervalMs;
+  private final String _zkConnect;
   private final ZkUtils _zkUtils;
   private final ScheduledExecutorService _executor;
   private volatile boolean _running = false;
   private final String _serviceName;
+  private final int _topicReplicationFactor;
+  private final boolean _topicCreationEnabled;
 
 
   public TopicManagementService(Map<String, Object> props, String serviceName) {
     _serviceName = serviceName;
     TopicManagementServiceConfig config = new TopicManagementServiceConfig(props);
     _partitionToBrokerRatioThreshold = config.getDouble(TopicManagementServiceConfig.PARTITIONS_TO_BROKER_RATIO_THRESHOLD);
     _topic = config.getString(CommonServiceConfig.TOPIC_CONFIG);
-    _partitionsToBrokerRatio = config.getDouble(CommonServiceConfig.PARTITIONS_TO_BROKER_RATO_CONFIG);
+    _partitionsToBrokerRatio = config.getDouble(TopicManagementServiceConfig.PARTITIONS_TO_BROKER_RATIO_CONFIG);
     _scheduleIntervalMs = config.getInt(TopicManagementServiceConfig.REBALANCE_INTERVAL_MS_CONFIG);
-    String zkUrl = config.getString(CommonServiceConfig.ZOOKEEPER_CONNECT_CONFIG);
-    _zkUtils = ZkUtils.apply(zkUrl, ZK_SESSION_TIMEOUT_MS, ZK_CONNECTION_TIMEOUT_MS, JaasUtils.isZkSecurityEnabled());
+    _zkConnect = config.getString(CommonServiceConfig.ZOOKEEPER_CONNECT_CONFIG);
+    _zkUtils = ZkUtils.apply(_zkConnect, ZK_SESSION_TIMEOUT_MS, ZK_CONNECTION_TIMEOUT_MS, JaasUtils.isZkSecurityEnabled());
     _executor = Executors.newSingleThreadScheduledExecutor(new TopicManagementServiceThreadFactory());
+    _topicReplicationFactor = config.getInt(TopicManagementServiceConfig.TOPIC_REPLICATION_FACTOR_CONFIG);
+    _topicCreationEnabled = config.getBoolean(TopicManagementServiceConfig.TOPIC_CREATION_ENABLED_CONFIG);
+
+    int existingPartitionCount = Utils.getPartitionNumForTopic(_zkConnect, _topic);
+    if (existingPartitionCount <= 0) {
+      if (_topicCreationEnabled) {
+        Utils.createMonitoringTopicIfNotExists(_zkConnect, _topic, _topicReplicationFactor, _partitionsToBrokerRatio);
```

Comment:
Do you actually need two places to have this logic of topic creation? Why should this be put?